setup.sh

```diff
@@ -21,7 +22,11 @@ if [ ! -e $env ]; then
 fi
 
 echo "=== Install Python packages into $env..."
-$env/bin/pip install sqlalchemy alembic pyyaml watchdog || exit 1
+$env/bin/pip install -r $codalabdir/requirements.txt || exit 1
+
+# Use sckoo@stanford.edu's patched argcomplete, until pull request has been accepted
+# https://github.com/kislyuk/argcomplete/pull/118
+$env/bin/pip install git+https://github.com/kashizui/argcomplete.git@allow-overridden-subparsersaction
```

Comment:
In codalab/codalab/settings/local.py, BUNDLE_SERVICE_CODE_PATH points to
the codalab-cli.


On Wed, Oct 28, 2015 at 12:53 AM, Stephen Koo <notifications@github.com>
wrote:

> In setup.sh
> <https://github.com/codalab/codalab-cli/pull/188#discussion_r43225303>:
>
> > @@ -21,7 +22,11 @@ if [ ! -e $env ]; then
> >  fi
> >
> >  echo "=== Install Python packages into $env..."
> > -$env/bin/pip install sqlalchemy alembic pyyaml watchdog || exit 1
> > +$env/bin/pip install -r $codalabdir/requirements.txt || exit 1
> > +
> > +# Use sckoo@stanford.edu's patched argcomplete, until pull request has been accepted
> > +# https://github.com/kislyuk/argcomplete/pull/118
> > +$env/bin/pip install git+https://github.com/kashizui/argcomplete.git@allow-overridden-subparsersaction
>
> Ohh... that's interesting.
> By the way could you explain or point me to documentation about how
> codalab is running code from my codalab-cli repo?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/codalab/codalab-cli/pull/188/files#r43225303>.
>